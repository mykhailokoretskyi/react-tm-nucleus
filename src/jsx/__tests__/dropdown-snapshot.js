import React from 'react';
import Dropdown from '../dropdown';
import renderer from 'react-test-renderer';

describe('Dropdown', () => {
    it('with value matches snapshot', () => {
        const component = renderer.create(
            <Dropdown defaultValue="a" >
                <option value="" />
                <option value="a">A</option>
                <option value="b">B</option>
            </Dropdown>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('with label matches snapshot', () => {
        const component = renderer.create(
            <Dropdown label="Choose" icon='<use xlink:href="img/someIcon" />'>
                <option value="" />
                <option value="a">A</option>
                <option value="b">B</option>
            </Dropdown>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('multiple matches snapshot', () => {
        const component = renderer.create(
            <Dropdown label="Choose" multiple={true} required={true} defaultValue={['b']} icon='<use xlink:href="img/someIcon" />'>
                <option value="" />
                <option value="a">A</option>
                <option value="b">B</option>
            </Dropdown>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
