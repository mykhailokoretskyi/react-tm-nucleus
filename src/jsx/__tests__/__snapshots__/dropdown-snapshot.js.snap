exports[`Dropdown multiple matches snapshot 1`] = `
<div
  className="input-group dropdown">
  <select
    className="dropdown__select"
    id=""
    multiple={true}
    onChange={[Function bound _onChange]}
    required={true}
    value={
      Array [
        "b"
      ]
    }>
    <option
      value="" />
    <option
      value="a">
      A
    </option>
    <option
      value="b">
      B
    </option>
  </select>
  <label
    className="input-group__label"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Choose&nbsp;<span class=\"input-group__required\">*</span>"
      }
    }
    htmlFor="" />
  <svg
    className="dropdown__icon-chevron"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<use xlink:href=\"img/someIcon\" />"
      }
    } />
</div>
`;

exports[`Dropdown with label matches snapshot 1`] = `
<div
  className="input-group dropdown">
  <select
    className="dropdown__select"
    id=""
    multiple={false}
    onChange={[Function bound _onChange]}
    required={false}
    value="">
    <option
      value="" />
    <option
      value="a">
      A
    </option>
    <option
      value="b">
      B
    </option>
  </select>
  <label
    className="input-group__label"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Choose"
      }
    }
    htmlFor="" />
  <svg
    className="dropdown__icon-chevron"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<use xlink:href=\"img/someIcon\" />"
      }
    } />
</div>
`;

exports[`Dropdown with value matches snapshot 1`] = `
<div
  className="input-group dropdown">
  <select
    className="dropdown__select"
    id=""
    multiple={false}
    onChange={[Function bound _onChange]}
    required={false}
    value="a">
    <option
      value="" />
    <option
      value="a">
      A
    </option>
    <option
      value="b">
      B
    </option>
  </select>
  <label
    className="input-group__label"
    dangerouslySetInnerHTML={
      Object {
        "__html": ""
      }
    }
    htmlFor="" />
  <svg
    className="dropdown__icon-chevron"
    dangerouslySetInnerHTML={
      Object {
        "__html": ""
      }
    } />
</div>
`;
